pkg/gardenadm/cmd/connect/connect.go

```diff
@@ -46,7 +64,111 @@ gardenadm connect`,
 	return cmd
 }
 
-func run(_ context.Context, opts *Options) error {
-	opts.Log.Info("Command is work in progress")
+func run(ctx context.Context, opts *Options) error {
+	opts.Log.Info("Using resources from directory", "configDir", opts.ConfigDir)
+
+	b, err := botanist.NewAutonomousBotanistFromManifests(ctx, opts.Log, nil, opts.ConfigDir, true)
+	if err != nil {
+		return fmt.Errorf("failed creating autonomous botanist: %w", err)
+	}
+	b.SeedClientSet, err = b.CreateClientSet(ctx)
+	if err != nil {
+		return fmt.Errorf("failed creating client set for autonomous shoot: %w", err)
+	}
+
+	gardenClientSet, err := kubernetes.NewWithConfig(kubernetes.WithRESTConfig(&rest.Config{
+		Host:            opts.ControlPlaneAddress,
+		TLSClientConfig: rest.TLSClientConfig{CAData: opts.CertificateAuthority},
+		BearerToken:     opts.BootstrapToken,
+	}), kubernetes.WithClientOptions(client.Options{Scheme: kubernetes.GardenScheme}), kubernetes.WithDisabledCachedClient())
+	if err != nil {
+		return fmt.Errorf("failed creating garden client set: %w", err)
+	}
+	b.GardenClient = gardenClientSet.Client()
+
+	gardenletDeployer := newGardenletDeployer(b, gardenClientSet)
+
+	var (
+		g = flow.NewGraph("connect")
+
+		_ = g.Add(flow.Task{
+			Name: "Deploying gardenlet into autonomous shoot cluster",
+			Fn: func(ctx context.Context) error {
+				_, err := gardenletDeployer.Reconcile(
+					ctx,
+					b.Logger,
+					b.Shoot.GetInfo(),
+					nil,
+					&seedmanagementv1alpha1.GardenletDeployment{},
+					&runtime.RawExtension{Object: &gardenletconfigv1alpha1.GardenletConfiguration{}},
+					seedmanagementv1alpha1.BootstrapToken,
+					false,
+				)
+				return err
+			},
+		})
+	)
+
+	if err := g.Compile().Run(ctx, flow.Opts{
+		Log: opts.Log,
+	}); err != nil {
+		return flow.Errors(err)
+	}
+
+	fmt.Fprintf(opts.Out, `
+Your autonomous shoot cluster has successfully been connected to Gardener!
+
+The gardenlet has been deployed in the %s namespace of your autonomous shoot
+cluster and is now taking over the management and lifecycle of it. All
+modifications to the Shoot specification should now be performed via the Gardener
+API, rather than by directly editing resources in the cluster.
+
+The bootstrap token will be deleted automatically by kube-controller-manager
+after it has expired. If you want to delete it right awy, run the following
+on any control plane node:
+
+  gardenadm token delete %s
+
+Resources have been successfully synchronized with the garden cluster. You may
+(and should) now remove them from the directory at %s, as they will eventually
+become outdated:
+
+  rm -rf %[3]s
```

Comment:
What's with the format string `%[3]s`? Why not just use `%s` instead?
I may be missing a crucial `Printf(...)` trick here...

Reply:
Ok, I missed one of the `%s` in the long format string. Thanks for the explanation.
As the explicit argument index feels like a niche feature, I would rather prefer to duplicate the parameter for readability, but have no strong opinion here.
WDYT?